compiler/rustc_mir/src/transform/check_unsafety.rs

```diff
@@ -580,24 +580,23 @@ fn is_enclosed(
     tcx: TyCtxt<'_>,
     used_unsafe: &FxHashSet<hir::HirId>,
     id: hir::HirId,
-) -> Option<(String, hir::HirId)> {
+    unsafe_fn_allowed: bool,
+) -> Option<(&'static str, hir::HirId)> {
     let parent_id = tcx.hir().get_parent_node(id);
     if parent_id != id {
         if used_unsafe.contains(&parent_id) {
-            Some(("block".to_string(), parent_id))
+            Some(("block", parent_id))
         } else if let Some(Node::Item(&hir::Item {
             kind: hir::ItemKind::Fn(ref sig, _, _), ..
         })) = tcx.hir().find(parent_id)
         {
-            if sig.header.unsafety == hir::Unsafety::Unsafe
-                && !tcx.features().unsafe_block_in_unsafe_fn
-            {
-                Some(("fn".to_string(), parent_id))
+            if sig.header.unsafety == hir::Unsafety::Unsafe && unsafe_fn_allowed {
+                Some(("fn", parent_id))
```

Comment:
What I am wondering... is this branch even reachable with `warn(unsafe_op_in_unsafe_fn)`? I think it should not be since an `unsafe` block directly in a `fn` will never be unused then.

So, why do you even need the `&& unsafe_fn_allowed`? If you remove it, does anything change?

Reply:
```rust
#[warn(unsafe_op_in_unsafe_fn)]
unsafe fn foo() {
    unsafe {}
}
```

If I remove the condition, I think the label will appear but it clezrly shouldn't.

Reply:
Oh, good point, I had not considered empty `unsafe` blocks.

Reply:
Maybe add a test for that?